### ISSUE TITLE
Rootfs mount order issue (tmpfs + zfs)

### ISSUE THREAD

**Issue by yangm97** [nix-community/disko#270]
## The problem

disko mounts zfs pool before the rootfs mount (which is tmpfs in my case), shadowing the zfs native mounts. Example log:

```
disko --mode mount hosts/sisyphus/disko-config.nix
+ zpool list sisyphus                                                                                                                                                                                             
+ zpool import -l -R /mnt sisyphus                                                                                                                                                                                
0 / 0 keys successfully loaded                                                                                                                                                                                    
+ findmnt tmpfs /mnt/
+ mount -t tmpfs none /mnt/ -o mode=755 -o X-mount.mkdir
+ findmnt /dev/disk/by-id/usb-USB_SanDisk_3.2Gen1_0901b343af36a38f56915e6ae7f6378d8e795333704cbe0c39e64a2528c00d94de2e000000000000000000003430e0f8ff1f2d2067558107c92aa62d-0:0-part1 /mnt/boot
+ mount /dev/disk/by-id/usb-USB_SanDisk_3.2Gen1_0901b343af36a38f56915e6ae7f6378d8e795333704cbe0c39e64a2528c00d94de2e000000000000000000003430e0f8ff1f2d2067558107c92aa62d-0:0-part1 /mnt/boot -t vfat -o defaults -o X-mount.mkdir
+ findmnt sisyphus/safe/persist /mnt/persist
+ mount sisyphus/safe/persist /mnt/persist -o X-mount.mkdir -o defaults -t zfs
```

## Ideal scenario

disko should mount the root before importing the zfs pool. 

BTW I'm only using one legacy mount to workaround impermance desire for a mount which has `neededForBoot` set from the nix configuration (so it doesn't quite "see" zfs native mounts).

## Workarounds

From the log output, it appears to me that using legacy mounts for all zfs datasets would work.

**Comment by Lassulus** [comment]
not sure if thats a disko or a nixpkgs problem. maybe posting a config which causes the breakage will help me investigate

**Comment by Enzime** [comment]
I think this issue should be fixed by #474

**Comment by iFreilicht** [comment]
As that PR got merged, I'll close this issue for now. Feel free to re-open if you're still experiencing this issue.
